src/modules/dashboard/components/Dialogs/EscrowFundsDialog/EscrowFundsDialog.tsx

```diff
@@ -0,0 +1,309 @@
+                              name="filteredDomainId"
+                              renderActiveOptionFn={renderActiveOption}
+                              filterOptionsFn={filterDomains}
+                              showAllDomains
+                              showDescription
+                              dataTest="colonyDomainSelector"
+                              itemDataTest="colonyDomainSelectorItem"
+                            />
+                          </span>
+                        )
+                      )}
+                    </div>
+                  </FormSection>
+                  <FormSection appearance={{ border: 'top' }}>
+                    <div className={styles.sectionRow} id={styles.balanceRow}>
+                      <InputLabel
+                        label={MSG.allocationBalance}
+                        appearance={{
+                          direction: 'horizontal',
+                        }}
+                      />
+                      <div>
+                        {balanceOptions.map((balanceOption) => {
+                          return (
+                            <div key={balanceOption.value}>
+                              {balanceOption.children}
+                            </div>
+                          );
+                        })}
+                      </div>
+                    </div>
+                  </FormSection>
+                </DialogSection>
+                <DialogSection appearance={{ border: 'bottom' }}>
+                  <Heading
+                    appearance={{
+                      size: 'small',
+                      weight: 'bold',
+                      theme: 'dark',
+                      margin: 'small',
+                    }}
+                  >
+                    <div className={styles.requiredFundsHeader}>
+                      <FormattedMessage {...MSG.fundsHeader} />
+                    </div>
+                  </Heading>
+                  {requiredFunds.map((balanceOption) => {
+                    return (
+                      <span key={balanceOption.value}>
+                        {balanceOption.children}
+                      </span>
+                    );
+                  })}
+                </DialogSection>
+                <DialogSection>
+                  <div className={styles.annotations}>
+                    <Annotations
+                      label={MSG.textareaLabel}
+                      name="annotationMessage"
+                    />
+                  </div>
+                </DialogSection>
+              </div>
+              <DialogSection appearance={{ align: 'right', theme: 'footer' }}>
+                <Button
+                  appearance={{
+                    theme: 'primary',
+                    size: 'large',
+                  }}
+                  autoFocus
+                  onClick={handleSubmit}
+                  text={MSG.confirmText}
```

Comment:
The button text should be conditional if we force it (action) or not (motion). It might be a miss in the specs, but please confirm with @arrenv, which copy to use for action.

Reply:
This change is recent, and was not included in their initial spec. It would be good to include it, but will require a rebase off of master.

@agnieszkajarosikloj let me know your thoughts on this. If not, please let me know so I can track it, if so, the logic for this can be found here - https://github.com/JoinColony/colonyDapp/blob/4a75beee2e86942fc1d1e29563ec3813896df0cd/src/modules/dashboard/components/Dialogs/UnlockTokenDialog/UnlockTokenForm.tsx#L186

Reply:
Do all the changes from master have to be added to the feature branch to add this functionality?
I'm afraid rebase can cause some conflicts and therefore take some time. 
Maybe after closing all branches from phase 1, we can do rebase? It won't reduce conflicts, but we could finish phase 1 first.

Reply:
Sounds good to me. I will create an issue to keep track of it. Thanks.

Reply:
Issue has been created here - https://github.com/JoinColony/colonyDapp/issues/3661